DigitalLearningSolutions.Web/Controllers/SignpostingController.cs

```diff
@@ -51,9 +51,11 @@ public async Task<IActionResult> LaunchLearningResource(int resourceReferenceId)
             }
 
             var (resource, apiIsAccessible) =
-                await learningHubResourceService.GetResourceByReferenceId(resourceReferenceId);
+                await learningHubResourceService.GetResourceByReferenceIdAndPopulateDeletedDetailsFromDatabase(
+                    resourceReferenceId
+                );
 
-            if (resource == null)
+            if (resource == null || resource.AbsentInLearningHub)

```

Comment:
I added this check because if the user hadn't dismissed the warning, this page would load for a removed resource, but we want it to return not found right? I've now moved this check earlier to before the redirection to ViewResource, because I think that method would still lead to the account linking url even if the resource is deleted, so this seemed the simplest fix. But let me know if that change is a bad idea for any reason

Reply:
I think this change is what we need. Given that the warning page is just a warning page about account linking, dismissing it should not really produce different functional results to not having it dismissed. So a Not Found in either case is best. This looks like a good solution to me.